version: '3.8'
services:
  python-job:
    build:
      context: .
      dockerfile: docker/python/Dockerfile
    volumes:
      - json-volume:/data  # Volume exclusivo para os JSON gerados
    environment:
      - PYTHON_ENV=production
    restart: unless-stopped
    depends_on:
      - mongodb
    networks:
      omie_net:
        ipv4_address: 172.28.0.2

  mongodb:
    image: mongo:latest
    container_name: mongodb
    volumes:
      - mongo-data:/data/db  # Volume exclusivo para os dados do MongoDB
    ports:
      - "27017:27017"
    restart: unless-stopped
    healthcheck:
      test: ["CMD", "mongo", "--eval", "db.adminCommand('ping')"]
      interval: 30s
      timeout: 10s
      retries: 3
    networks:
      omie_net:
        ipv4_address: 172.28.0.3

  postgres:
    image: postgres:latest
    container_name: postgres
    environment:
      - POSTGRES_USER=omieapi
      - POSTGRES_PASSWORD="admin123%"
      - POSTGRES_DB=db_omieoximag
    ports:
      - "5432:5432"
    volumes:
      - pgdata:/var/lib/postgresql/data
    command: ["postgres", "-c", "listen_addresses='*'"]
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U omieapi"]
      interval: 30s
      timeout: 10s
      retries: 3
    networks:
      omie_net:
        ipv4_address: 172.28.0.4

volumes:
  json-volume:
  mongo-data:
  pgdata:

networks:
  omie_net:
    driver: bridge
    ipam:
      config:
        - subnet: 172.28.0.0/16
